### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by arthcp**
Changes unit tests according to predicate design. 
Has following tests -
- Set values on `RootChainManager`.
- Lock `ERC20` on root chain.
- Lock `ERC721` on root chain.
- Set values on `ChildChainManager`.
- Mint `ERC20` on child chain.

### CODE REVIEW

**Comment by jdkanani** [comment]
@arthcp Can we add truffle test coverage the same as contracts repo?

File: package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "migrate:4": "truffle migrate --network root --f 4 --to 4",
     "migrate:5": "truffle migrate --network child --f 5 --to 5",
     "migrate": "npm run migrate:2 && npm run migrate:3 && npm run migrate:4 && npm run migrate:5",
-    "test": "truffle test",
+    "test": "truffle test --network root --bail",
```
Comment (jdkanani) [inline package.json:17]: What `--bail` does?

File: test/root/RootChainManager.test.js [hunk 2]
```diff
@@ -155,74 +182,73 @@ contract('RootChainManager', async(accounts) => {
...
-        const lockedLogAmount = new BN(lockedLog.args.amount.toString())
-        lockedLogAmount.should.be.bignumber.that.equals(depositAmount)
+      it('tokenId', () => {
+        const lockedLogTokenId = lockedLog.args.tokenId.toNumber()
+        lockedLogTokenId.should.equal(depositTokenId)
       })
 
-      it('user', () => {
-        lockedLog.args.user.should.equal(depositForAccount)
+      it('depositReceiver', () => {
```
Comment (jdkanani) [inline test/root/RootChainManager.test.js:246]:
```suggestion
      it('should emit valid deposit receiver', () => {
```

File: test/root/RootChainManager.test.js [hunk 3]
```diff
@@ -155,74 +182,73 @@ contract('RootChainManager', async(accounts) => {
...
         lockedLog.address.should.equal(
-          rootChainManager.address.toLowerCase()
+          contracts.root.erc721Predicate.address.toLowerCase()
         )
       })
 
-      it('amount', () => {
-        const lockedLogAmount = new BN(lockedLog.args.amount.toString())
-        lockedLogAmount.should.be.bignumber.that.equals(depositAmount)
+      it('tokenId', () => {
```
Comment (jdkanani) [inline test/root/RootChainManager.test.js:241]:
```suggestion
      it('should set token id properly', () => {
```

File: test/root/RootChainManager.test.js [hunk 4]
```diff
@@ -155,74 +182,73 @@ contract('RootChainManager', async(accounts) => {
...
-      await dummyToken.approve(rootChainManager.address, depositAmount)
-      depositTx = await rootChainManager.depositFor(depositForAccount, dummyToken.address, depositAmount)
+      await dummyERC721.approve(contracts.root.erc721Predicate.address, depositTokenId)
+      const depositData = abi.encode(['uint256'], [depositTokenId.toString()])
+      depositTx = await rootChainManager.depositFor(depositForAccount, dummyERC721.address, depositData)
       should.exist(depositTx)
     })
 
-    it('Emits Locked log', () => {
+    it('Emits LockedERC721 log', () => {
```
Comment (jdkanani) [inline test/root/RootChainManager.test.js:228]:
```suggestion
    it('should emit LockedERC721 log', () => {
```

File: test/root/RootChainManager.test.js [hunk 5]
```diff
@@ -155,74 +182,73 @@ contract('RootChainManager', async(accounts) => {
...
+      dummyERC721 = contracts.root.dummyERC721
       rootChainManager = contracts.root.rootChainManager
-      oldAccountBalance = await dummyToken.balanceOf(accounts[0])
-      oldContractBalance = await dummyToken.balanceOf(rootChainManager.address)
+      await dummyERC721.mint(depositTokenId)
     })
 
-    it('Account has balance', () => {
-      depositAmount.should.be.a.bignumber.lessThan(oldAccountBalance)
+    it('Account has token', async() => {
```
Comment (jdkanani) [inline test/root/RootChainManager.test.js:216]:
```suggestion
    it('should set proper token balance to account', async() => {
```

File: package.json (hunk 1 above)
Comment (arthcp) [reply package.json:17]: `--bail` stops test execution after first failed test

**Pull request merged**